File: pdi/docs/First_steps/hello_expose.jl
struct PC_tree_t
    # The tree status
    status::Cint
    # The document containing the tree
    document::Ptr{Cvoid}
    # the node inside the tree
    node ::Ptr{Cvoid}
end

# These are the values for enum PDI_inout_t 
PDI_NONE  = 0
PDI_IN    = 1
PDI_OUT   = 2
PDI_INOUT = 3

conf = @ccall "libparaconf".PC_parse_path("hello_data.yml"::Cstring)::PC_tree_t

@ccall "libpdi".PDI_init(conf::PC_tree_t)::Cvoid

my_world = Int32(42)
# [share_reclaim]
@ccall "libpdi".PDI_share("world"::Cstring, my_world::Cint, PDI_OUT::Cint)::Cvoid
@ccall "libpdi".PDI_reclaim("world"::Cstring)::Cvoid
# [share_reclaim]

# [expose]
@ccall "libpdi".PDI_expose("world"::Cstring, my_world::Cint,PDI_OUT::Cint)::Cvoid
# [expose]

@ccall "libpdi".PDI_finalize()::Cvoid

File: pdi/docs/First_steps/hello_expose.c
/*
 * SPDX-FileCopyrightText: 2021-2026 Commissariat a l'energie atomique et aux energies alternatives (CEA)
 * SPDX-FileCopyrightText: 2024 Commissariat a l'energie atomique et aux energies alternatives (CEA)
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

#include <pdi.h>

int main(int argc, char* argv[])
{
	PDI_init(PC_parse_path("hello_data.yml"));
	int my_world = 42;

	//! [share_reclaim]
	PDI_share("world", &my_world, PDI_OUT);
	PDI_reclaim("world");
	//! [share_reclaim]

	//! [expose]
	PDI_expose("world", &my_world, PDI_OUT);
	//! [expose]

	PDI_finalize();
	return 0;
}
